src/components/onboarding/recover-account-form/index.tsx

```diff
@@ -0,0 +1,400 @@
+import { useState } from 'react';
+import styled from 'styled-components';
+import toast from 'react-hot-toast';
+import { Button } from '@/components/generic/button';
+import { useForm } from '@/components/generic/form/use-form';
+import { FormInput } from '@/components/generic/form/form-input';
+import { FormSelect } from '@/components/generic/form/form-select';
+import { networkList } from '@/constants/network-list';
+import { Row } from '@/components/generic/row';
+import { useAppDispatch, useAppSelector } from '@/hooks';
+import { OnboardingHeading } from '@/components/onboarding/onboarding-header';
+import { clearState } from '@/features/onboarding-slice';
+import { navigateTo } from '@/utils/navigation';
+import { routes } from '@/constants/routes';
+import { db } from '@/db';
+import { ButtonWrapper } from '@/generic/button-wrapper';
+import { FormTextArea } from '@/components/generic/form/form-text-area';
+import { isBackgroundPageResponding } from '@/utils/communication';
+import { FormCheckbox } from '@/components/generic/form/form-checkbox';
+import { AnchorLink } from '@/components/anchor-link';
+import { WalletIcon } from '@/svg/wallet-icon';
+import { Chevron } from '@/components/svg/chevron';
+import { EyeOffIcon } from '@/components/svg/eye-off-icon';
+
+import { isValidMnemonic } from '@/utils/blockchain';
+import type { CreateAccountReturnType } from '@/utils/account-factory';
+
+type Props = {
+  className?: string;
+  step: 1 | 2 | 3;
```

Comment:
can we use something like:
```
stepName: "accountType" | "accountForm" | "accountPassword";
``` 
Names just as example. Probably you could think of better ones :)